### ISSUE TITLE
Incorrect updating of wt-at-age vector at beginning of year

### ISSUE THREAD

**Issue by Rick-Methot-NOAA** [nmfs-stock-synthesis/stock-synthesis#138]
This bug was detected while testing code for issue #102 .  The bug is described with this new issue.

The problem stems from the need to have smrybiomass, stored in Smry_Table(y,2), available before any time-varying parameters are applied.  This is because of possibility that a time-varying parameter will invoke density-dependence, hence need access to the current summary biomass.
When growth is updated for a year using the growth2() function, it projects mean length-at-age to the beginning of the next year. This vector of mean lengths can is used to calculate wt_age_beg(s,g)  which is then used to calculate smrybiomass.  That appears to be working OK, but a double-check now seems warranted.
The real issue is with empirical wt-at-age.  It is used to fill wt_age_beg(s,g), but not until later in the calculation cycle.  So when smrybiomass is calculated for the year, it uses a version of wt_age_beg that was filled the previous year.  This was not obvious because the value for smrybiomass reported in the time_series output table is re-calculated at report time, so uses the correct year's wtatage.  However, if the user has selected to report F_std as exploitation rate in biomass, then it uses the flawed version of smrybiomass.  The consequence is that if:
a) empirical wt-at-age is used
b)  body wt in that input varies by year
and
c)  F_std is based on catch/smrybiomass
then the F_std values will be incorrect.
This has now been corrected, but a much more streamlined and less error-prone approach is available.

NEW APPROACH to be done:
a.  change wt_age_beg(s,g,a) to wt_age_beg(t,g,a);  where t is index to continuous seasons throughout the year. This obviates need to constantly move values to and from save_wt_age with code like:
    Save_Wt_Age(t)=Wt_Age_beg(s);
b.  upon reading empirical weight, move info directly into wt_age_beg(t,g,a) in the prelim calcs.  This removes the need to move the info on every iteration!!!!!!
    if(WTage_rd>0)
.......
        Wt_Age_beg(s,g)=WTage_emp(t,GP3(g),0);

concordant changes can happen for Wt_age_mid and for fecundity.

**Comment by Rick-Methot-NOAA** [comment]
this appears to be resolved.
note that a substantial re-work of the storage of such quantities and their order of calculation is in the new-natmort branch, but further work is on hold pending higher priorities
